plugins/ckan_operators/package_operator.py

```diff
@@ -212,4 +213,86 @@ def execute(self, context):
             if self.lists_match( self.package_a, self.package_b ):
                 return {"result": True}
 
-        # TODO: add logic for comparing only packages and not lists of packages
\ No newline at end of file
+        # TODO: add logic for comparing only packages and not lists of packages
+
+class GetCkanPackageListOperator(BaseOperator):
+    """
+    Get full package list
+    """
+    @apply_defaults
+    def __init__(
+        self,
+        address: str,
+        apikey: str,
+        **kwargs
+    ) -> None:
+        super().__init__(**kwargs)
+        self.ckan = ckanapi.RemoteCKAN(apikey=apikey, address=address)
+
+
+    def execute(self, context):
+        package_list = self.ckan.action.package_list()
+        logging.info(f'package list {package_list}')
+        return package_list
+
+class CheckCkanResourceDescriptionOperator(BaseOperator):
+    """
+    Check if the description of a resource is missing
+    """
+    @apply_defaults
+    def __init__(
+        self,
+        input_dag_id: str = None,
+        base_url : str = None,
+        resource_url : str = None,
+        **kwargs
+    ) -> None:
+        super().__init__(**kwargs)
+        self.input_dag_id = input_dag_id
+        self.base_url = base_url
```

Comment:
for both `base_url` and `resource_url`, dont feed in hardcode URLs here. If you do, this DAG will call DEV CKAN even when we deploy it to PROD Airflow!

Use, instead, something like below to grab the base CKAN url, and then modify that as you need:
https://github.com/open-data-toronto/operations/blob/b918d84cf9a3c694b8155949b691761ca0cba01f/dags/datasets/files_to_datastore/files_to_datastore.py#L33

Reply:
ok, will modify